modules/client/src/client.ts

```diff
@@ -3,43 +3,36 @@ import { ClientMethods } from "./internal/client/methods";
 import { ClientEncoding } from "./internal/client/encoding";
 import { ClientEstimation } from "./internal/client/estimation";
 import { ClientDecoding } from "./internal/client/decoding";
-import {
-  IClient,
-  IClientDecoding,
-  IClientEncoding,
-  IClientEstimation,
-  IClientMethods,
-} from "./interfaces";
 import { ClientCore } from "./client-common/core";
 
 /**
  * Provider a generic client with high level methods to manage and interact with DAO's
  */
-export class Client extends ClientCore implements IClient {
-  private privateMethods: IClientMethods;
-  private privateEncoding: IClientEncoding;
-  private privateDecoding: IClientDecoding;
-  private privateEstimation: IClientEstimation;
+export class Client extends ClientCore {
+  private privateMethods: ClientMethods;
```

Comment:
I'm surprised this works, because passing a class as a type, expects the value to be a class... and not an instance. 

I've had problems with this and my solution has been to use `typeof ClientMethods` to represent instances

Reply:
work flawlessly. you only need to use `typeof` when you got out of the typing environment.
For example, if you wanna know if the const `client` is of type `Client`.
But given this are internal these checks aren't necessary and therefore we can use the class as type

Reply:
Right, one thing is `typeof` used in run time to get a value, but it has a completely different use on the TypeScript land. 

Anyway if these types are used only internally, then there's no problem here :+1: